src/threading/formatters/Ascii.cc

```diff
@@ -325,6 +325,28 @@ threading::Value* Ascii::ParseValue(const string& s, const string& name, TypeTag
 		break;
 		}
 
+	case TYPE_PATTERN:
+		{
+		string cannidate = get_unescaped_string(s);
+		// A string is a cannidate pattern iff it begins and ends with
+		// a '/'. Rather or not the rest of the string is legal will
+		// be determined later when it is given to the RE engine.
+		if ( cannidate.size() >= 2 )
+			{
+				if ( cannidate.front() == cannidate.back() &&
+					cannidate.back() == '/' )
+					{
+					// Remove the '/'s
+					cannidate.erase(0, 1);
+					cannidate.erase(cannidate.size() - 1);
+					val->val.pattern_text_val = copy_string(cannidate.c_str());
```

Comment:
Looks like a memory leak since `copy_string` allocates, but the `~Value()` destructor has not been made aware of how to clean up this new `pattern_text_val` union member.

Related to previous comment about potentially re-using the `string_val` member instead, looks like you'd still need to adapt the dtor to clean up memory for pattern types in that case.

Generally might be a good opportunity to add a memory leak test for this (or alter an existing one, maybe `testing/btest/core/leaks/input-basic.zeek`).  E.g. might adapt the test first, see if it reports a leak, then fix it.

Reply:
Shoot. Thanks for catching that. I'll go ahead and add the test and see if I can catch the leak then fix it up.